cmd/metacache-server-pool.go

```diff
@@ -189,11 +189,16 @@ func (z *erasureServerPools) listPath(ctx context.Context, o listPathOptions) (e
 					if err != nil {
 						return true
 					}
-					oFIV, err := existing.fileInfo(o.Bucket)
+					oFIV, err := other.fileInfo(o.Bucket)
 					if err != nil {
 						return false
 					}
-					return oFIV.ModTime.After(eFIV.ModTime)
+					// Replace if modtime is newer
+					if !oFIV.ModTime.Equal(eFIV.ModTime) {
+						return oFIV.ModTime.After(eFIV.ModTime)
+					}
+					// Use NumVersions as a final tiebreaker.
+					return oFIV.NumVersions > eFIV.NumVersions
```

Comment:
I don't think this is needed, modtime is in nanosecond, better to panic() because this means something is seriously broken here

Reply:
@vadmeste This is expected to have collisions since reads are unlocked and at different spots in time. We just use the information available.

Reply:
> @vadmeste This is expected to have collisions since reads are unlocked and at different spots in time. We just use the information available.

I mean I find it hard to have two files with the same bucket and same name to be have the same nanotime modtime.